extensions/wikia/MercuryApi/handlers/MercuryApiFilePageHandler.class.php

```diff
@@ -12,8 +12,25 @@ public static function getFileContent( Title $title ) {
 		$details = WikiaFileHelper::getMediaDetail( $title );
 		$mediaObject = ArticleAsJson::createMediaObject( $details, $title->getText() );
 
+		// if article contains user provided HTML which is invalid (e.g. to many </div>), snippetter treat it as a text
+		// and it is not removed as other tags. In mobile-wiki we do not escape snippets, therefore invalid html may
+		// cause errors on front-end side
+		$fileUsageList = array_map( function( $item ) {
```

Comment:
should we check if fileUsageList is an array?

Reply:
we don't have to, $fileUsageData['fileList'] is always an array and array_map returns array if array provided as source